R/repr_recordedplot.r

```diff
@@ -87,13 +104,11 @@ repr_png.recordedplot <- function(obj,
 	res       = getOption('repr.plot.res'),
 ...) {
 	if (!is_cairo_installed() && !check_capability('png')) return(NULL)
-	
-	dev.cb <- function(tf)
-		if (is_cairo_installed())
-			Cairo::Cairo(width, height, tf, 'png', pointsize, bg, 'transparent', 'in', res)
-		else
-			png(tf, width, height, 'in', pointsize, bg, res, antialias = antialias)
-	
+	dev.cb <- function(tf) get_device(
+		ragg = ragg::agg_png(tf, width, height, 'in', pointsize, bg, res),  # scaling, bitsize
```

Comment:
Looking at the code, at first I would have expected this to cause issues if the `ragg` package was not present on the end-user system, since I'd expect the namespace to not be known. Previously, for Cairo, you had it wrapped in if-else statemtents, so then the Cairo namespace would clearly only be called if the `is_cairo_installed()` returned true.

However, _I_ initially didn't have ragg, and this line didn't produce any issues. I'm not R expert, so just want to check: are you sure this will work ok, even if some of the backends are absent? And which function now imports this namespace? Before, I believe that was done by the `requireNamespace` call that was done by e.g. `is_cairo_installed()`, but now, it's not clear to me.